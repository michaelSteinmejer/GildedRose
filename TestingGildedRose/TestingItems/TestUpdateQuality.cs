using System;
using System.CodeDom;
using System.Collections.Generic;
using ApprovalTests.Combinations;
using ApprovalTests.Reporters;
using GildedRose.Model;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace TestingGildedRose.TestingItems
{
    [TestClass]
    public class TestUpdateQuality
    {   /// <summary>
        ///In this test method, i have implemented the dependency ApprovalTest.
        ///It makes it easy, to test many scenarios and get the full CodeCoverage.
        ///the idea is that VerifyAllCombination passes arguments to a method,
        ///until all combinations are met. It then generates two text files Received.txt
        ///and approved.txt, you then approve the cases in the Received.txt by
        ///copying it to approved.txt.
        /// </summary>
        [TestMethod]
        [UseReporter(typeof(DiffReporter))]
        public void TestingUpdateQuality(){

            CombinationApprovals.VerifyAllCombinations(DoUpdateQuality,
                new string[] {
                    "Aged Brie",
                    "Elixir of the Mongoose",
                    "Sulfuras, Hand of Ragnaros",
                    "+5 Dexterity Vest",
                    "Backstage passes to a TAFKAL80ETC concert",
                    "Conjured Mana Cake"
                },
                new int[] { -1, 0, 5, 10, 15, 25 },
                new int[] { 0, 1, 10, 20, 49, 55 }

                );
        }

        private static string DoUpdateQuality(string name, int sellIn, int quality)
        {
            var items = new List<Item>{new Item{Name=name,SellIn=sellIn,Quality=quality} };
             GildedRose.Controller.ItemUpdateHandler.UpdateQuality(1, items);
            return items[0].ToString();
        }
    }
}
