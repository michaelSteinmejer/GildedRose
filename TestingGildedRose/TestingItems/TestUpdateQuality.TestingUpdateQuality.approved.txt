[Aged Brie,-1,0] => Name: Aged Brie, SellIn: -2, Quality: 2
[Aged Brie,-1,1] => Name: Aged Brie, SellIn: -2, Quality: 3
[Aged Brie,-1,10] => Name: Aged Brie, SellIn: -2, Quality: 12
[Aged Brie,-1,20] => Name: Aged Brie, SellIn: -2, Quality: 22
[Aged Brie,-1,49] => Name: Aged Brie, SellIn: -2, Quality: 50
[Aged Brie,-1,55] => Name: Aged Brie, SellIn: -2, Quality: 50
[Aged Brie,0,0] => Name: Aged Brie, SellIn: -1, Quality: 2
[Aged Brie,0,1] => Name: Aged Brie, SellIn: -1, Quality: 3
[Aged Brie,0,10] => Name: Aged Brie, SellIn: -1, Quality: 12
[Aged Brie,0,20] => Name: Aged Brie, SellIn: -1, Quality: 22
[Aged Brie,0,49] => Name: Aged Brie, SellIn: -1, Quality: 50
[Aged Brie,0,55] => Name: Aged Brie, SellIn: -1, Quality: 50
[Aged Brie,5,0] => Name: Aged Brie, SellIn: 4, Quality: 1
[Aged Brie,5,1] => Name: Aged Brie, SellIn: 4, Quality: 2
[Aged Brie,5,10] => Name: Aged Brie, SellIn: 4, Quality: 11
[Aged Brie,5,20] => Name: Aged Brie, SellIn: 4, Quality: 21
[Aged Brie,5,49] => Name: Aged Brie, SellIn: 4, Quality: 50
[Aged Brie,5,55] => Name: Aged Brie, SellIn: 4, Quality: 50
[Aged Brie,10,0] => Name: Aged Brie, SellIn: 9, Quality: 1
[Aged Brie,10,1] => Name: Aged Brie, SellIn: 9, Quality: 2
[Aged Brie,10,10] => Name: Aged Brie, SellIn: 9, Quality: 11
[Aged Brie,10,20] => Name: Aged Brie, SellIn: 9, Quality: 21
[Aged Brie,10,49] => Name: Aged Brie, SellIn: 9, Quality: 50
[Aged Brie,10,55] => Name: Aged Brie, SellIn: 9, Quality: 50
[Aged Brie,15,0] => Name: Aged Brie, SellIn: 14, Quality: 1
[Aged Brie,15,1] => Name: Aged Brie, SellIn: 14, Quality: 2
[Aged Brie,15,10] => Name: Aged Brie, SellIn: 14, Quality: 11
[Aged Brie,15,20] => Name: Aged Brie, SellIn: 14, Quality: 21
[Aged Brie,15,49] => Name: Aged Brie, SellIn: 14, Quality: 50
[Aged Brie,15,55] => Name: Aged Brie, SellIn: 14, Quality: 50
[Aged Brie,25,0] => Name: Aged Brie, SellIn: 24, Quality: 1
[Aged Brie,25,1] => Name: Aged Brie, SellIn: 24, Quality: 2
[Aged Brie,25,10] => Name: Aged Brie, SellIn: 24, Quality: 11
[Aged Brie,25,20] => Name: Aged Brie, SellIn: 24, Quality: 21
[Aged Brie,25,49] => Name: Aged Brie, SellIn: 24, Quality: 50
[Aged Brie,25,55] => Name: Aged Brie, SellIn: 24, Quality: 50
[Elixir of the Mongoose,-1,0] => Name: Elixir of the Mongoose, SellIn: -2, Quality: 0
[Elixir of the Mongoose,-1,1] => Name: Elixir of the Mongoose, SellIn: -2, Quality: 0
[Elixir of the Mongoose,-1,10] => Name: Elixir of the Mongoose, SellIn: -2, Quality: 8
[Elixir of the Mongoose,-1,20] => Name: Elixir of the Mongoose, SellIn: -2, Quality: 18
[Elixir of the Mongoose,-1,49] => Name: Elixir of the Mongoose, SellIn: -2, Quality: 47
[Elixir of the Mongoose,-1,55] => Name: Elixir of the Mongoose, SellIn: -2, Quality: 50
[Elixir of the Mongoose,0,0] => Name: Elixir of the Mongoose, SellIn: -1, Quality: 0
[Elixir of the Mongoose,0,1] => Name: Elixir of the Mongoose, SellIn: -1, Quality: 0
[Elixir of the Mongoose,0,10] => Name: Elixir of the Mongoose, SellIn: -1, Quality: 8
[Elixir of the Mongoose,0,20] => Name: Elixir of the Mongoose, SellIn: -1, Quality: 18
[Elixir of the Mongoose,0,49] => Name: Elixir of the Mongoose, SellIn: -1, Quality: 47
[Elixir of the Mongoose,0,55] => Name: Elixir of the Mongoose, SellIn: -1, Quality: 50
[Elixir of the Mongoose,5,0] => Name: Elixir of the Mongoose, SellIn: 4, Quality: 0
[Elixir of the Mongoose,5,1] => Name: Elixir of the Mongoose, SellIn: 4, Quality: 0
[Elixir of the Mongoose,5,10] => Name: Elixir of the Mongoose, SellIn: 4, Quality: 9
[Elixir of the Mongoose,5,20] => Name: Elixir of the Mongoose, SellIn: 4, Quality: 19
[Elixir of the Mongoose,5,49] => Name: Elixir of the Mongoose, SellIn: 4, Quality: 48
[Elixir of the Mongoose,5,55] => Name: Elixir of the Mongoose, SellIn: 4, Quality: 50
[Elixir of the Mongoose,10,0] => Name: Elixir of the Mongoose, SellIn: 9, Quality: 0
[Elixir of the Mongoose,10,1] => Name: Elixir of the Mongoose, SellIn: 9, Quality: 0
[Elixir of the Mongoose,10,10] => Name: Elixir of the Mongoose, SellIn: 9, Quality: 9
[Elixir of the Mongoose,10,20] => Name: Elixir of the Mongoose, SellIn: 9, Quality: 19
[Elixir of the Mongoose,10,49] => Name: Elixir of the Mongoose, SellIn: 9, Quality: 48
[Elixir of the Mongoose,10,55] => Name: Elixir of the Mongoose, SellIn: 9, Quality: 50
[Elixir of the Mongoose,15,0] => Name: Elixir of the Mongoose, SellIn: 14, Quality: 0
[Elixir of the Mongoose,15,1] => Name: Elixir of the Mongoose, SellIn: 14, Quality: 0
[Elixir of the Mongoose,15,10] => Name: Elixir of the Mongoose, SellIn: 14, Quality: 9
[Elixir of the Mongoose,15,20] => Name: Elixir of the Mongoose, SellIn: 14, Quality: 19
[Elixir of the Mongoose,15,49] => Name: Elixir of the Mongoose, SellIn: 14, Quality: 48
[Elixir of the Mongoose,15,55] => Name: Elixir of the Mongoose, SellIn: 14, Quality: 50
[Elixir of the Mongoose,25,0] => Name: Elixir of the Mongoose, SellIn: 24, Quality: 0
[Elixir of the Mongoose,25,1] => Name: Elixir of the Mongoose, SellIn: 24, Quality: 0
[Elixir of the Mongoose,25,10] => Name: Elixir of the Mongoose, SellIn: 24, Quality: 9
[Elixir of the Mongoose,25,20] => Name: Elixir of the Mongoose, SellIn: 24, Quality: 19
[Elixir of the Mongoose,25,49] => Name: Elixir of the Mongoose, SellIn: 24, Quality: 48
[Elixir of the Mongoose,25,55] => Name: Elixir of the Mongoose, SellIn: 24, Quality: 50
[Sulfuras, Hand of Ragnaros,-1,0] => Name: Sulfuras, Hand of Ragnaros, SellIn: -1, Quality: 0
[Sulfuras, Hand of Ragnaros,-1,1] => Name: Sulfuras, Hand of Ragnaros, SellIn: -1, Quality: 1
[Sulfuras, Hand of Ragnaros,-1,10] => Name: Sulfuras, Hand of Ragnaros, SellIn: -1, Quality: 10
[Sulfuras, Hand of Ragnaros,-1,20] => Name: Sulfuras, Hand of Ragnaros, SellIn: -1, Quality: 20
[Sulfuras, Hand of Ragnaros,-1,49] => Name: Sulfuras, Hand of Ragnaros, SellIn: -1, Quality: 49
[Sulfuras, Hand of Ragnaros,-1,55] => Name: Sulfuras, Hand of Ragnaros, SellIn: -1, Quality: 55
[Sulfuras, Hand of Ragnaros,0,0] => Name: Sulfuras, Hand of Ragnaros, SellIn: 0, Quality: 0
[Sulfuras, Hand of Ragnaros,0,1] => Name: Sulfuras, Hand of Ragnaros, SellIn: 0, Quality: 1
[Sulfuras, Hand of Ragnaros,0,10] => Name: Sulfuras, Hand of Ragnaros, SellIn: 0, Quality: 10
[Sulfuras, Hand of Ragnaros,0,20] => Name: Sulfuras, Hand of Ragnaros, SellIn: 0, Quality: 20
[Sulfuras, Hand of Ragnaros,0,49] => Name: Sulfuras, Hand of Ragnaros, SellIn: 0, Quality: 49
[Sulfuras, Hand of Ragnaros,0,55] => Name: Sulfuras, Hand of Ragnaros, SellIn: 0, Quality: 55
[Sulfuras, Hand of Ragnaros,5,0] => Name: Sulfuras, Hand of Ragnaros, SellIn: 5, Quality: 0
[Sulfuras, Hand of Ragnaros,5,1] => Name: Sulfuras, Hand of Ragnaros, SellIn: 5, Quality: 1
[Sulfuras, Hand of Ragnaros,5,10] => Name: Sulfuras, Hand of Ragnaros, SellIn: 5, Quality: 10
[Sulfuras, Hand of Ragnaros,5,20] => Name: Sulfuras, Hand of Ragnaros, SellIn: 5, Quality: 20
[Sulfuras, Hand of Ragnaros,5,49] => Name: Sulfuras, Hand of Ragnaros, SellIn: 5, Quality: 49
[Sulfuras, Hand of Ragnaros,5,55] => Name: Sulfuras, Hand of Ragnaros, SellIn: 5, Quality: 55
[Sulfuras, Hand of Ragnaros,10,0] => Name: Sulfuras, Hand of Ragnaros, SellIn: 10, Quality: 0
[Sulfuras, Hand of Ragnaros,10,1] => Name: Sulfuras, Hand of Ragnaros, SellIn: 10, Quality: 1
[Sulfuras, Hand of Ragnaros,10,10] => Name: Sulfuras, Hand of Ragnaros, SellIn: 10, Quality: 10
[Sulfuras, Hand of Ragnaros,10,20] => Name: Sulfuras, Hand of Ragnaros, SellIn: 10, Quality: 20
[Sulfuras, Hand of Ragnaros,10,49] => Name: Sulfuras, Hand of Ragnaros, SellIn: 10, Quality: 49
[Sulfuras, Hand of Ragnaros,10,55] => Name: Sulfuras, Hand of Ragnaros, SellIn: 10, Quality: 55
[Sulfuras, Hand of Ragnaros,15,0] => Name: Sulfuras, Hand of Ragnaros, SellIn: 15, Quality: 0
[Sulfuras, Hand of Ragnaros,15,1] => Name: Sulfuras, Hand of Ragnaros, SellIn: 15, Quality: 1
[Sulfuras, Hand of Ragnaros,15,10] => Name: Sulfuras, Hand of Ragnaros, SellIn: 15, Quality: 10
[Sulfuras, Hand of Ragnaros,15,20] => Name: Sulfuras, Hand of Ragnaros, SellIn: 15, Quality: 20
[Sulfuras, Hand of Ragnaros,15,49] => Name: Sulfuras, Hand of Ragnaros, SellIn: 15, Quality: 49
[Sulfuras, Hand of Ragnaros,15,55] => Name: Sulfuras, Hand of Ragnaros, SellIn: 15, Quality: 55
[Sulfuras, Hand of Ragnaros,25,0] => Name: Sulfuras, Hand of Ragnaros, SellIn: 25, Quality: 0
[Sulfuras, Hand of Ragnaros,25,1] => Name: Sulfuras, Hand of Ragnaros, SellIn: 25, Quality: 1
[Sulfuras, Hand of Ragnaros,25,10] => Name: Sulfuras, Hand of Ragnaros, SellIn: 25, Quality: 10
[Sulfuras, Hand of Ragnaros,25,20] => Name: Sulfuras, Hand of Ragnaros, SellIn: 25, Quality: 20
[Sulfuras, Hand of Ragnaros,25,49] => Name: Sulfuras, Hand of Ragnaros, SellIn: 25, Quality: 49
[Sulfuras, Hand of Ragnaros,25,55] => Name: Sulfuras, Hand of Ragnaros, SellIn: 25, Quality: 55
[+5 Dexterity Vest,-1,0] => Name: +5 Dexterity Vest, SellIn: -2, Quality: 0
[+5 Dexterity Vest,-1,1] => Name: +5 Dexterity Vest, SellIn: -2, Quality: 0
[+5 Dexterity Vest,-1,10] => Name: +5 Dexterity Vest, SellIn: -2, Quality: 8
[+5 Dexterity Vest,-1,20] => Name: +5 Dexterity Vest, SellIn: -2, Quality: 18
[+5 Dexterity Vest,-1,49] => Name: +5 Dexterity Vest, SellIn: -2, Quality: 47
[+5 Dexterity Vest,-1,55] => Name: +5 Dexterity Vest, SellIn: -2, Quality: 50
[+5 Dexterity Vest,0,0] => Name: +5 Dexterity Vest, SellIn: -1, Quality: 0
[+5 Dexterity Vest,0,1] => Name: +5 Dexterity Vest, SellIn: -1, Quality: 0
[+5 Dexterity Vest,0,10] => Name: +5 Dexterity Vest, SellIn: -1, Quality: 8
[+5 Dexterity Vest,0,20] => Name: +5 Dexterity Vest, SellIn: -1, Quality: 18
[+5 Dexterity Vest,0,49] => Name: +5 Dexterity Vest, SellIn: -1, Quality: 47
[+5 Dexterity Vest,0,55] => Name: +5 Dexterity Vest, SellIn: -1, Quality: 50
[+5 Dexterity Vest,5,0] => Name: +5 Dexterity Vest, SellIn: 4, Quality: 0
[+5 Dexterity Vest,5,1] => Name: +5 Dexterity Vest, SellIn: 4, Quality: 0
[+5 Dexterity Vest,5,10] => Name: +5 Dexterity Vest, SellIn: 4, Quality: 9
[+5 Dexterity Vest,5,20] => Name: +5 Dexterity Vest, SellIn: 4, Quality: 19
[+5 Dexterity Vest,5,49] => Name: +5 Dexterity Vest, SellIn: 4, Quality: 48
[+5 Dexterity Vest,5,55] => Name: +5 Dexterity Vest, SellIn: 4, Quality: 50
[+5 Dexterity Vest,10,0] => Name: +5 Dexterity Vest, SellIn: 9, Quality: 0
[+5 Dexterity Vest,10,1] => Name: +5 Dexterity Vest, SellIn: 9, Quality: 0
[+5 Dexterity Vest,10,10] => Name: +5 Dexterity Vest, SellIn: 9, Quality: 9
[+5 Dexterity Vest,10,20] => Name: +5 Dexterity Vest, SellIn: 9, Quality: 19
[+5 Dexterity Vest,10,49] => Name: +5 Dexterity Vest, SellIn: 9, Quality: 48
[+5 Dexterity Vest,10,55] => Name: +5 Dexterity Vest, SellIn: 9, Quality: 50
[+5 Dexterity Vest,15,0] => Name: +5 Dexterity Vest, SellIn: 14, Quality: 0
[+5 Dexterity Vest,15,1] => Name: +5 Dexterity Vest, SellIn: 14, Quality: 0
[+5 Dexterity Vest,15,10] => Name: +5 Dexterity Vest, SellIn: 14, Quality: 9
[+5 Dexterity Vest,15,20] => Name: +5 Dexterity Vest, SellIn: 14, Quality: 19
[+5 Dexterity Vest,15,49] => Name: +5 Dexterity Vest, SellIn: 14, Quality: 48
[+5 Dexterity Vest,15,55] => Name: +5 Dexterity Vest, SellIn: 14, Quality: 50
[+5 Dexterity Vest,25,0] => Name: +5 Dexterity Vest, SellIn: 24, Quality: 0
[+5 Dexterity Vest,25,1] => Name: +5 Dexterity Vest, SellIn: 24, Quality: 0
[+5 Dexterity Vest,25,10] => Name: +5 Dexterity Vest, SellIn: 24, Quality: 9
[+5 Dexterity Vest,25,20] => Name: +5 Dexterity Vest, SellIn: 24, Quality: 19
[+5 Dexterity Vest,25,49] => Name: +5 Dexterity Vest, SellIn: 24, Quality: 48
[+5 Dexterity Vest,25,55] => Name: +5 Dexterity Vest, SellIn: 24, Quality: 50
[Backstage passes to a TAFKAL80ETC concert,-1,0] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -2, Quality: 0
[Backstage passes to a TAFKAL80ETC concert,-1,1] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -2, Quality: 0
[Backstage passes to a TAFKAL80ETC concert,-1,10] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -2, Quality: 0
[Backstage passes to a TAFKAL80ETC concert,-1,20] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -2, Quality: 0
[Backstage passes to a TAFKAL80ETC concert,-1,49] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -2, Quality: 0
[Backstage passes to a TAFKAL80ETC concert,-1,55] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -2, Quality: 0
[Backstage passes to a TAFKAL80ETC concert,0,0] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -1, Quality: 0
[Backstage passes to a TAFKAL80ETC concert,0,1] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -1, Quality: 0
[Backstage passes to a TAFKAL80ETC concert,0,10] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -1, Quality: 0
[Backstage passes to a TAFKAL80ETC concert,0,20] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -1, Quality: 0
[Backstage passes to a TAFKAL80ETC concert,0,49] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -1, Quality: 0
[Backstage passes to a TAFKAL80ETC concert,0,55] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: -1, Quality: 0
[Backstage passes to a TAFKAL80ETC concert,5,0] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 4, Quality: 3
[Backstage passes to a TAFKAL80ETC concert,5,1] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 4, Quality: 4
[Backstage passes to a TAFKAL80ETC concert,5,10] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 4, Quality: 13
[Backstage passes to a TAFKAL80ETC concert,5,20] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 4, Quality: 23
[Backstage passes to a TAFKAL80ETC concert,5,49] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 4, Quality: 50
[Backstage passes to a TAFKAL80ETC concert,5,55] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 4, Quality: 50
[Backstage passes to a TAFKAL80ETC concert,10,0] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 9, Quality: 2
[Backstage passes to a TAFKAL80ETC concert,10,1] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 9, Quality: 3
[Backstage passes to a TAFKAL80ETC concert,10,10] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 9, Quality: 12
[Backstage passes to a TAFKAL80ETC concert,10,20] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 9, Quality: 22
[Backstage passes to a TAFKAL80ETC concert,10,49] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 9, Quality: 50
[Backstage passes to a TAFKAL80ETC concert,10,55] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 9, Quality: 50
[Backstage passes to a TAFKAL80ETC concert,15,0] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 14, Quality: 1
[Backstage passes to a TAFKAL80ETC concert,15,1] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 14, Quality: 2
[Backstage passes to a TAFKAL80ETC concert,15,10] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 14, Quality: 11
[Backstage passes to a TAFKAL80ETC concert,15,20] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 14, Quality: 21
[Backstage passes to a TAFKAL80ETC concert,15,49] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 14, Quality: 50
[Backstage passes to a TAFKAL80ETC concert,15,55] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 14, Quality: 50
[Backstage passes to a TAFKAL80ETC concert,25,0] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 24, Quality: 1
[Backstage passes to a TAFKAL80ETC concert,25,1] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 24, Quality: 2
[Backstage passes to a TAFKAL80ETC concert,25,10] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 24, Quality: 11
[Backstage passes to a TAFKAL80ETC concert,25,20] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 24, Quality: 21
[Backstage passes to a TAFKAL80ETC concert,25,49] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 24, Quality: 50
[Backstage passes to a TAFKAL80ETC concert,25,55] => Name: Backstage passes to a TAFKAL80ETC concert, SellIn: 24, Quality: 50
[Conjured Mana Cake,-1,0] => Name: Conjured Mana Cake, SellIn: -2, Quality: 0
[Conjured Mana Cake,-1,1] => Name: Conjured Mana Cake, SellIn: -2, Quality: 0
[Conjured Mana Cake,-1,10] => Name: Conjured Mana Cake, SellIn: -2, Quality: 6
[Conjured Mana Cake,-1,20] => Name: Conjured Mana Cake, SellIn: -2, Quality: 16
[Conjured Mana Cake,-1,49] => Name: Conjured Mana Cake, SellIn: -2, Quality: 45
[Conjured Mana Cake,-1,55] => Name: Conjured Mana Cake, SellIn: -2, Quality: 50
[Conjured Mana Cake,0,0] => Name: Conjured Mana Cake, SellIn: -1, Quality: 0
[Conjured Mana Cake,0,1] => Name: Conjured Mana Cake, SellIn: -1, Quality: 0
[Conjured Mana Cake,0,10] => Name: Conjured Mana Cake, SellIn: -1, Quality: 6
[Conjured Mana Cake,0,20] => Name: Conjured Mana Cake, SellIn: -1, Quality: 16
[Conjured Mana Cake,0,49] => Name: Conjured Mana Cake, SellIn: -1, Quality: 45
[Conjured Mana Cake,0,55] => Name: Conjured Mana Cake, SellIn: -1, Quality: 50
[Conjured Mana Cake,5,0] => Name: Conjured Mana Cake, SellIn: 4, Quality: 0
[Conjured Mana Cake,5,1] => Name: Conjured Mana Cake, SellIn: 4, Quality: 0
[Conjured Mana Cake,5,10] => Name: Conjured Mana Cake, SellIn: 4, Quality: 8
[Conjured Mana Cake,5,20] => Name: Conjured Mana Cake, SellIn: 4, Quality: 18
[Conjured Mana Cake,5,49] => Name: Conjured Mana Cake, SellIn: 4, Quality: 47
[Conjured Mana Cake,5,55] => Name: Conjured Mana Cake, SellIn: 4, Quality: 50
[Conjured Mana Cake,10,0] => Name: Conjured Mana Cake, SellIn: 9, Quality: 0
[Conjured Mana Cake,10,1] => Name: Conjured Mana Cake, SellIn: 9, Quality: 0
[Conjured Mana Cake,10,10] => Name: Conjured Mana Cake, SellIn: 9, Quality: 8
[Conjured Mana Cake,10,20] => Name: Conjured Mana Cake, SellIn: 9, Quality: 18
[Conjured Mana Cake,10,49] => Name: Conjured Mana Cake, SellIn: 9, Quality: 47
[Conjured Mana Cake,10,55] => Name: Conjured Mana Cake, SellIn: 9, Quality: 50
[Conjured Mana Cake,15,0] => Name: Conjured Mana Cake, SellIn: 14, Quality: 0
[Conjured Mana Cake,15,1] => Name: Conjured Mana Cake, SellIn: 14, Quality: 0
[Conjured Mana Cake,15,10] => Name: Conjured Mana Cake, SellIn: 14, Quality: 8
[Conjured Mana Cake,15,20] => Name: Conjured Mana Cake, SellIn: 14, Quality: 18
[Conjured Mana Cake,15,49] => Name: Conjured Mana Cake, SellIn: 14, Quality: 47
[Conjured Mana Cake,15,55] => Name: Conjured Mana Cake, SellIn: 14, Quality: 50
[Conjured Mana Cake,25,0] => Name: Conjured Mana Cake, SellIn: 24, Quality: 0
[Conjured Mana Cake,25,1] => Name: Conjured Mana Cake, SellIn: 24, Quality: 0
[Conjured Mana Cake,25,10] => Name: Conjured Mana Cake, SellIn: 24, Quality: 8
[Conjured Mana Cake,25,20] => Name: Conjured Mana Cake, SellIn: 24, Quality: 18
[Conjured Mana Cake,25,49] => Name: Conjured Mana Cake, SellIn: 24, Quality: 47
[Conjured Mana Cake,25,55] => Name: Conjured Mana Cake, SellIn: 24, Quality: 50
